libs/api/protocol/src/lib/mapper-profile/protocol.mapper-profile.ts

```diff
@@ -0,0 +1,96 @@
+import type { Mapper } from '@automapper/core';
+import { createMap, forMember, mapFrom } from '@automapper/core';
+import { AutomapperProfile, getMapperToken } from '@automapper/nestjs';
+import { Inject, Injectable } from '@nestjs/common';
+
+import { UserProducer } from '../core/entity/partials/producer-partial.entity';
+import {
+	CommunicationMessage,
+	CommunicationMessagePayload,
+} from '../core/entity/protocol-entries/communication-message.entity';
+import {
+	CommunicationMessageDocument,
+	CommunicationMessagePayloadDocument,
+} from '../infra/schema/communication-message.schema';
+import { UserProducerDocument } from '../infra/schema/producer-partial.schema';
+import {
+	unitDocumentToEntityMapper,
+	unitEntityToDocumentMapper,
+} from './unit-partial.mapper-profile';
+
+@Injectable()
+export class CommunicationMessageProfile extends AutomapperProfile {
+	constructor(@Inject(getMapperToken()) mapper: Mapper) {
+		super(mapper);
+	}
+
+	override get profile() {
+		return (mapper: Mapper): void => {
+			// document --> entity
+			createMap(
+				mapper,
+				CommunicationMessagePayloadDocument,
+				CommunicationMessagePayload,
+			);
+			createMap(
```

Comment:
we have the BaseMapperProfile, which maps the default properties of our base document and base entity model. Currently, i strictly introduced one profile per direction (document -> entity, entity->document, dto->document...), what are your thoughts on this?

Reply:
I think this is very verbose and might lead to a lot of profile classes. However, we can simply go with that rule for now and later evaluate if we want to group them differently as refactoring seems very simple.

Reply:
Alright, sounds good. I will refactor my working branches towards this structure